File: app/src/main/java/com/satnamsinghmaggo/paathapp/model/Bani.java
package com.satnamsinghmaggo.paathapp.model;

import android.os.Parcel;
import android.os.Parcelable;

public class Bani implements Parcelable {
    private String name;


    public Bani(String name) {
        this.name = name;

    }

    protected Bani(Parcel in) {
        name = in.readString();
    }

    public static final Creator<Bani> CREATOR = new Creator<Bani>() {
        @Override
        public Bani createFromParcel(Parcel in) {
            return new Bani(in);
        }

        @Override
        public Bani[] newArray(int size) {
            return new Bani[size];
        }
    };

    public String getName() {
        return name;
    }


    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(name);
    }

    @Override
    public int describeContents() {
        return 0;
    }
}

File: app/src/main/java/com/satnamsinghmaggo/paathapp/model/Bani.kt
package com.satnamsinghmaggo.paathapp.model

import android.os.Parcel
import android.os.Parcelable

data class Bani(val name: String, val time: String) : Parcelable {
    constructor(parcel: Parcel) : this(
        parcel.readString() ?: "",
        parcel.readString() ?: ""
    )

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(name)
        parcel.writeString(time)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Bani> {
        override fun createFromParcel(parcel: Parcel): Bani {
            return Bani(parcel)
        }

        override fun newArray(size: Int): Array<Bani?> {
            return arrayOfNulls(size)
        }
    }
} 